include/powsybl/math/UndirectedGraph.hxx

```diff
@@ -81,6 +93,25 @@ unsigned long UndirectedGraph<V, E>::addVertex() {
     return v;
 }
 
+template <typename V, typename E>
+void UndirectedGraph<V, E>::addVertexIfNotPresent(unsigned long v) {
+    if (v < m_vertices.size()) {
+        const auto& it = m_availableVertices.find(v);
+        if (it != m_availableVertices.end()) {
+            m_vertices[v] = stdcxx::make_unique<Vertex>();
+            m_availableVertices.erase(it);
+        }
+    } else {
+        std::size_t oldSize = m_vertices.size();
+        m_vertices.resize(v + 1);
```

Comment:
Moving this line after the for loop will allow you to remove the oldSize variable, and write:
```
        for (unsigned int i = m_vertices.size(); i < v; ++i) {
            m_availableVertices.insert(m_availableVertices.end(), i);
        }
        m_vertices.resize(v + 1);
```
I don't understand why you try to add the value at the end of the set, using a position, and not simply insert values in the set: `m_availableVertices.insert(i);`

Reply:
Agree... I though it was more optimized, but I forgot that we are pushing decreasingly :sweat_smile: